src/components/activity/Activity.tsx

```diff
@@ -27,29 +39,102 @@ const Activity = ({
     startTime,
     endTime,
     categoryIds,
+    isTemplate,
+    isSelected,
+    onSelectedActivityChange,
+    templateView,
+    onCloseTemplateModal,
 }: ActivityComponentProps) => {
     const { categories } = useContext(CategoriesContext);
     const [isEditFormOpen, setIsEditFormOpen] = useState<boolean>(false);
 
-    const handleOnClick = () => setIsEditFormOpen(true);
+    const { setSelectedDay } = useContext(DayViewContext);
+
+    const activityRef = useRef<HTMLElement>(null);
 
     const associatedCategories: Category[] | undefined = categories?.filter(
         (category) =>
             category && category.id ? categoryIds.includes(category.id) : false
     );
 
+    const handleOnFormOpen = (event: MouseEvent | TouchEvent) => {
+        event.stopPropagation();
+        setIsEditFormOpen(true);
+        onSelectedActivityChange(0);
+    };
+
+    const handleOnTemplateChange = (event: MouseEvent | TouchEvent) => {
+        event.stopPropagation();
+        db.activities.update(id, { isTemplate: isTemplate === 1 ? 0 : 1 });
+    };
+
+    const handleOnDuplicateActivity = (event: MouseEvent | TouchEvent) => {
+        event.stopPropagation();
+        db.activities.add({
+            iconKey,
+            title,
+            description,
+            rating,
+            startTime: new Date(),
+            endTime: undefined,
+            categoryIds,
+        } as IActivity);
+        setSelectedDay(new Date());
+        onSelectedActivityChange(0);
+        if (onCloseTemplateModal) onCloseTemplateModal();
+    };
```

Comment:
I'd expect a duplicate action to open up a form with the same data already filled, rather than create a clone instantly. This makes 2 activities that happened at the exact same time, which should be invalid logically.